lib/utils/request.js

```diff
@@ -207,6 +208,26 @@ function MULTIGET (fhurls, where, cb) {
 }
 
 /**
+ * Creates a closure over a callback that handles error and status code checking for a upload request
+ * @param  {Function} cb Callback function
+ * @return {Function}      Node-style callback closed over `cb`
+ * @access private
+ */
+function uploadHandlerFor(cb) {
+  return function(err, response) {
+    if(err) {
+      return cb("Error Uploading File " + err);
+    }
+
+    if (response.statusCode.toString()[0]!=='2') {
+      return cb("Error - non 2xx status code: " + response.statusCode + " " + response.body);
```

Comment:
@vchepeli  I am not sure I agree. This code has been here for a long time and it is working fine. Why suddenly change it? Plus the response.body will provide more accurate information about the error. As as far as I know, the APIs will not return binary data for errors?